todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsFragment.java

```diff
@@ -13,87 +13,93 @@
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
 
 import com.example.android.architecture.blueprints.todoapp.R;
+import com.google.common.base.Preconditions;
 
-import static com.google.common.base.Preconditions.checkNotNull;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
 
 /**
  * Main UI for the statistics screen.
  */
-public class StatisticsFragment extends Fragment implements StatisticsContract.View {
+public class StatisticsFragment extends Fragment {
+
+    private static final String TAG = StatisticsFragment.class.getSimpleName();
 
     private TextView mStatisticsTV;
 
-    private StatisticsContract.Presenter mPresenter;
+    @Nullable
+    private StatisticsViewModel mViewModel;
+
+    @Nullable
+    private CompositeSubscription mSubscription;
 
     public static StatisticsFragment newInstance() {
         return new StatisticsFragment();
     }
 
-    @Override
-    public void setPresenter(@NonNull StatisticsContract.Presenter presenter) {
-        mPresenter = checkNotNull(presenter);
-    }
-
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View root = inflater.inflate(R.layout.statistics_frag, container, false);
         mStatisticsTV = (TextView) root.findViewById(R.id.statistics);
+
+        mViewModel = StatisticsModule.createStatisticsViewModel(getContext());
+
         return root;
     }
 
     @Override
     public void onResume() {
+        bindViewModel();
         super.onResume();
-        mPresenter.subscribe();
     }
 
     @Override
     public void onPause() {
+        unbindViewModel();
         super.onPause();
-        mPresenter.unsubscribe();
-    }
-
-    @Override
-    public void setProgressIndicator(boolean active) {
-        if (active) {
-            mStatisticsTV.setText(getString(R.string.loading));
-        }
     }
 
-    @Override
-    public void showStatistics(int numberOfIncompleteTasks, int numberOfCompletedTasks) {
-        if (numberOfCompletedTasks == 0 && numberOfIncompleteTasks == 0) {
-            mStatisticsTV.setText(getResources().getString(R.string.statistics_no_tasks));
-        } else {
-            String displayString = getResources().getString(R.string.statistics_active_tasks) + " "
-                    + numberOfIncompleteTasks + "\n" + getResources().getString(
-                    R.string.statistics_completed_tasks) + " " + numberOfCompletedTasks;
-            mStatisticsTV.setText(displayString);
-        }
+    private void bindViewModel() {
+        Preconditions.checkNotNull(mViewModel);
+        // using a CompositeSubscription to gather all the subscriptions, so all of them can be
+        // later unsubscribed together
```

Comment:
I know it's just a trifle, but could you change this comment to: 

```
// using a CompositeSubscription to gather all the subscriptions,
// so all of them can be later unsubscribed together
```
At least for me, it's more readable